src/sync/chase_lev.rs

```diff
@@ -245,19 +262,25 @@ impl<T> Deque<T> {
     // to be read after we flag this buffer for reclamation. All stealers,
     // however, have their own epoch pinned during this time so the buffer will
     // just naturally be free'd once all concurrent stealers have exited.
+    //
+    // This method may only be called from the workers, as it modifies the bottom
+    // counter.
     unsafe fn swap_buffer<'a>(&self,
                               b: isize,
                               old: Shared<'a, Buffer<T>>,
                               buf: Buffer<T>,
                               guard: &'a epoch::Guard)
                               -> Shared<'a, Buffer<T>> {
+        // replace the old buffer with the new.
         let newbuf = Owned::new(buf);
         let newbuf = self.array.store_and_ref(newbuf, SeqCst, &guard);
+        // add the new size to the bottom and top counters as a way of signalling
+        // to current stealers that the buffer has been resized.
         let ss = newbuf.size() as isize;
-        self.bottom.store(b + ss, SeqCst);
+        self.bottom.store(b + ss, Relaxed);
         let t = self.top.load(SeqCst);
         if self.top.compare_and_swap(t, t + ss, SeqCst) != t {
-            self.bottom.store(b, SeqCst);
+            self.bottom.store(b, Relaxed);
```

Comment:
The second paper follows the pattern that "bottom" is always modified with "Relaxed" orderings, which is acceptable since those modifications only ever occur on a single thread. Bottom is only read in "steal" other threads after a full memory barrier, so Relaxed should be sufficient here.

It is a cold path, though, so maybe you are right that it's best not to tempt fate by relaxing the orderings.